src/Init/System/IOError.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2019 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Simon Hudon
+-/
+
+prelude
+import Init.Core
+import Init.Data.UInt
+import Init.Data.ToString
+import Init.Data.String.Basic
+
+/-
+Immitate the structure of IOErrorType in Haskell:
+https://hackage.haskell.org/package/base-4.12.0.0/docs/System-IO-Error.html#t:IOErrorType
+-/
+inductive IO.Error
+| alreadyExists (details : String) -- EEXIST, EINPROGRESS, EISCONN
```

Comment:
Why store `details` as a `String`?

Reply:
Rather than as the `errno` code? Decoding the `errno` takes a IO action implemented in C. I thought I'd get all the decoding done at once on the C side.

Reply:
Not advocating for it but you could return to Lean an element of an enum type, e.g. `EINTR : ErrorCode`.

Reply:
I used to have that enumerated type on the Lean type but Leo and I decided to move it to the C side. What we could do instead is represent `EINTR` as a string. What do you think?